### PR TITLE
Support Apple Pay in API

### PR DESCRIPTION
**Pull request by jakubtobiasz**
Part of #45

### CODE REVIEW

File: tests/Api/Responses/shop/paying_for_orders_by_apple_pay/test_paying_with_a_valid_apple_pay_token_for_an_order.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  "@context": {
+    "@vocab": "http://localhost/api/v2/docs.jsonld#",
+    "hydra": "http://www.w3.org/ns/hydra/core#",
+    "status": "PayResult/status",
+    "transactionPaymentUrl": "PayResult/transactionPaymentUrl"
+  },
+  "@type": "PayResult",
+  "@id": "/api/v2/.well-known/genid/@string@",
+  "status": "pending",
+  "transactionPaymentUrl": "apple_pay://@string@"
```
Comment (jakubtobiasz) [inline tests/Api/Responses/shop/paying_for_orders_by_apple_pay/test_paying_with_a_valid_apple_pay_token_for_an_order.json:11]: It helps us to make sure we got a response for Apple Pay from Mockoon; we can add something similar for other payment methods in a separate PR

File: src/Payum/Action/Api/CreateApplePayTransactionAction.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            $this->createApplePayPaymentPayloadFactory->createFrom($model, $notifyToken->getTargetUrl(), $localeCode),
+        );
+
+        $paymentDetails = PaymentDetails::fromArray($model->getDetails());
+        $paymentDetails->setTransactionId($response['transactionId']);
+        $paymentDetails->setStatus($response['status']);
+
+        if ($this->is3dSecureRedirectRequired($paymentDetails)) {
+            $paymentDetails->setPaymentUrl(
+                $response['transactionPaymentUrl'] ?? throw new \InvalidArgumentException('Cannot perform 3DS redirect. Missing transactionPaymentUrl in the response.'),
```
Comment (lchrusciel) [inline src/Payum/Action/Api/CreateApplePayTransactionAction.php:48]: exception could be custom

File: src/Tpay/Factory/CreateApplePayPaymentPayloadFactory.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    ) {
+    }
+
+    public function createFrom(PaymentInterface $payment, string $notifyUrl, string $localeCode): array
+    {
+        /** @var array{pay: array<string, mixed>} $payload */
+        $payload = $this->createRedirectBasedPaymentPayloadFactory->createFrom($payment, $notifyUrl, $localeCode);
+
+        $paymentDetails = PaymentDetails::fromArray($payment->getDetails());
+        $applePayToken = $paymentDetails->getApplePayToken() ?? throw new \InvalidArgumentException('The given payment does not have an Apple Pay token.');
```
Comment (lchrusciel) [inline src/Tpay/Factory/CreateApplePayPaymentPayloadFactory.php:24]: another good candidate for custom exception

**Review by lchrusciel — APPROVED** [review]

File: .gitignore [hunk 4]
```diff
@@ -1,10 +1,11 @@
-/vendor/
+/cert/
```
[the remark this answers is not in the document]
Comment (jakubtobiasz) [reply .gitignore:1]: `/vendor/` is duplicated below :p

**Review by coldic3 — APPROVED** [review]

**Pull request merged**